Source/Extensions/Blazorise.Components/Autocomplete.razor.cs

```diff
@@ -176,21 +177,25 @@ protected async Task OnTextChangedHandler( string text )
                 }
                 else
                 {
-                    await ClearSelected();
+                    await ResetActiveItemIndex();
+                    await ResetSelectedValue();
+
+                    if ( FreeTyping )
+                    {
+                        selectedText = CurrentSearch;
+                        await SelectedTextChanged.InvokeAsync( selectedText );
+                    }
+                    else
+                    {
+                        await ResetSelectedText();
+                    }
                 }
             }
 
-            if ( HaveFilteredData )
-            {
-                ActiveItemIndex = 0;
-                await ScrollItemIntoView( ActiveItemIndex );
-            }
-            else
+            if ( !HaveFilteredData && !string.IsNullOrEmpty( CurrentSearch ) )
             {
-                if ( !string.IsNullOrEmpty( CurrentSearch ) )
-                {
-                    await NotFound.InvokeAsync( CurrentSearch );
-                }
+                // should fire in freetyping mode?
```

Comment:
You can remove the comment, yes you can just let it fire. Idea is everything works mostly as before, but with `FreeTyping` you can commit a value outside the `Data`.